src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1353,159 +1353,162 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
         dstCount = 1;
     }
 
-    const bool hasImmediateOperand = HWIntrinsicInfo::HasImmediateOperand(intrin.id);
-
-    if (hasImmediateOperand && !HWIntrinsicInfo::NoJmpTableImm(intrin.id))
-    {
-        // We may need to allocate an additional general-purpose register when an intrinsic has a non-const immediate
-        // operand and the intrinsic does not have an alternative non-const fallback form.
-        // However, for a case when the operand can take only two possible values - zero and one
-        // the codegen can use cbnz to do conditional branch, so such register is not needed.
+    // We may need to allocate an additional general-purpose register when an intrinsic has a non-const immediate
+    // operand and the intrinsic does not have an alternative non-const fallback form.
+    // However, for a case when the operand can take only two possible values - zero and one
+    // the codegen can use cbnz to do conditional branch, so such register is not needed.
+    auto needsBranchTargetReg = [this](const HWIntrinsic& intrinsic, const GenTreeHWIntrinsic* tree) -> bool {
```

Comment:
can you monitor the TP diffs because of this change?

Reply:
For this intrinsic, I think we should just special case it instead of wrapping common code paths in lamba.

Reply:
I'm not seeing any [diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=727057&view=ms.vss-build-web.run-extensions-tab), though I can undo this if you'd prefer less churn.